Meeting-15/README.md

```diff
@@ -1,28 +1,286 @@
-* [`<Router>`](https://reactrouter.com/web/api/Router)
-* [`<Link>`](https://reactrouter.com/web/api/Link)
-* [`<Switch>`](https://reactrouter.com/web/api/Switch)
-* [`<Route>`](https://reactrouter.com/web/api/Route)
-* [`<Redirect>`](https://reactrouter.com/web/api/Redirect)
-* API - Hooks
+- [Meeting 15](#meeting-15)
+  - [Table of Contents](#table-of-contents)
+  - [Instalation](#instalation)
+  - [Basic setup](#basic-setup)
+  - [Link component](#link-component)
+  - [NavLink](#navlink)
+  - [useNavigate()](#usenavigate)
+  - [Grouping Routes](#grouping-routes)
+  - [Dynamic routes](#dynamic-routes)
+  - [useLocation()](#uselocation)
+  - [Recomendations](#recomendations)
 
-  * [useHistory](https://reactrouter.com/web/api/Hooks/usehistory)
-  * [useLocation](https://reactrouter.com/web/api/Hooks/uselocation)
-  * [useParams](https://reactrouter.com/web/api/Hooks/useparams)
-  * [useRouteMatch](https://reactrouter.com/web/api/Hooks/useroutematch)
+React Router DOM is a collection of navigational components that compose declaratively with your application. It provides the components and APIs necessary to navigate through application screens, build dynamic navigation elements, and keep the URL in sync with what's being viewed on the page. It also provides an intuitive programming model for integrating with other libraries and frameworks, including React Navigation and Redux.
 
+## Instalation
 
-## References
+In your terminal run:
 
-[Routes :ReactRouter.com](https://reactrouter.com/web/guides/quick-start)
+```
+npm install react-router-dom
+```
 
+## Basic setup
 
-## Recomendations
+Import BrowserRouter, Routes, Route in your index.js and App.js files.
+
+index.js:
+
+```
+import {BrowserRouter} from 'react-router-dom'
+
+root.render(
+  <StrictMode>
+  	<BrowserRouter>
+	    <App />
+	<BrowserRouter/>
+  </StrictMode>
+);
+```
+
+App.js:
+
+```
+
+import {Routes, Route} from 'react-router-dom'
+
+export default function App() {
+  return (
+	<Routes>
+		<Route path="/" element={<Home />}>
+		<Route path="/products" element={<Products />}>
+		<Route path="/admin/dashboard" element={<Dashboard />}>
+	</Routes>
+  );
+}
+```
+
+**Takeaways**
+
+- Singla Page Application (SPA) is a type of modern web app where we need to download the static asets for once.
+- SPA utilises client-side routing for page navigation.
+- React router is a client-side routing library for React applications.
+- We use BrowserRouter, Routes and Route to register pages in our app.
+
+<Route> should be the ONLY children of <Routes>
+
+## Link component
+
+The Link component in React Router DOM is used to link different pages in a single page application. It is used to navigate between different routes in the application.
+
+Building a single page application means that we cant's use <a> tag to do our in-app navigation because the moment we click on an <a> tag the whole page will refresh and that defeats the purpose of a single page application.
```

Comment:
```suggestion
Building a single-page application means that we cant's use the `<a>` tag to do our in-app navigation because the moment we click on the `<a>` tag the whole page will refresh and that defeats the purpose of a single-page application.
```